mscr-ui/src/modules/workspace/group-home/index.tsx

```diff
@@ -27,25 +30,31 @@ export default function GroupWorkspace({
   pid,
   contentType,
 }: GroupHomeProps) {
-  const { t, i18n } = useTranslation('common');
+  const { t } = useTranslation('common');
   const { isSmall } = useBreakpoints();
-  const [currentPage, setCurrentPage] = useState(1);
+  const { urlState } = useUrlState();
+  const dispatch = useStoreDispatch();
   const pageSize = 20;
-  const query: PaginatedQuery = {
+  const { data, isLoading } = useGetOrgContentQuery({
     type: contentType,
-    ownerOrg: pid,
     pageSize,
-    pageFrom: (currentPage - 1) * pageSize,
-  };
-  const { data, isLoading } = useGetOrgContentQuery(query);
+    urlState,
+    ownerOrg: pid,
+  });
   const lastPage = data?.hits.total?.value
     ? Math.ceil(data?.hits.total.value / pageSize)
     : 0;
 
-   // Need to decide what data we want to fetch loading the application
   const refetchInfo = () => {
```

Comment:
For some reason, the cache invalidation doesn't get the content updated properly if it happens too fast. I suspect the indexing in the backend takes a bit, but I don't really know if it's that or something else.